jupyter_scheduler/scheduler.py

```diff
@@ -470,12 +530,41 @@ def list_job_definitions(self, query: ListJobDefinitionsQuery) -> ListJobDefinit
         return list_response
 
     def get_staging_paths(self, job_id: str) -> Dict[str, str]:
-        model = self.get_job(job_id)
+        model = None
+        with self.db_session() as session:
+            model = session.get(Job, job_id)
+
         staging_paths = {}
-        for output in model.outputs:
-            filename = create_output_filename(
-                model.input_uri, model.create_time, output.output_format
-            )
-            staging_paths[output.output_format] = os.path.join(self.staging_path, job_id, filename)
+        if model:
+            for output_format in model.output_formats:
+                filename = create_output_filename(model.input_uri, model.create_time, output_format)
+                staging_paths[output_format] = os.path.join(self.staging_path, job_id, filename)
+
+        return staging_paths
+
+
+class ArchiveDownloadingScheduler(Scheduler):
+    """Scheduler that adds archive path to staging paths.
+
+    Notes
+    -----
+    Should be used in tandem with :class:`~jupyter_scheduler.executors.ArchiveExportingExecutingManager`
+    during jupyter server start.
+    """
```

Comment:
This class should set its own default for the `execution_manager_class` trait. That way users don't need to manually don't need to manually specify the correct execution manager class in tandem.

Reply:
Good idea!